hikari/impl/rest_bot.py

```diff
@@ -0,0 +1,510 @@
+            rest_url=rest_url,
+            token=token,
+            token_type=token_type,
+        )
+
+        # IntegrationServer
+        self._server = interaction_server_impl.InteractionServer(
+            entity_factory=self._entity_factory,
+            public_key=public_key,
+            rest_client=self._rest,
+        )
+
+    @property
+    def is_alive(self) -> bool:
+        return self._server.is_alive
+
+    @property
+    def interaction_server(self) -> interaction_server_.InteractionServer:
+        return self._server
+
+    @property
+    def rest(self) -> rest_.RESTClient:
+        return self._rest
+
+    @property
+    def entity_factory(self) -> entity_factory_.EntityFactory:
+        return self._entity_factory
+
+    @property
+    def http_settings(self) -> config.HTTPSettings:
+        return self._http_settings
+
+    @property
+    def proxy_settings(self) -> config.ProxySettings:
+        return self._proxy_settings
+
+    @property
+    def executor(self) -> typing.Optional[concurrent.futures.Executor]:
+        return self._executor
+
+    @staticmethod
+    def print_banner(banner: typing.Optional[str], allow_color: bool, force_color: bool) -> None:
+        """Print the banner.
+
+        This allows library vendors to override this behaviour, or choose to
+        inject their own "branding" on top of what hikari provides by default.
+
+        Normal users should not need to invoke this function, and can simply
+        change the `banner` argument passed to the constructor to manipulate
+        what is displayed.
+
+        Parameters
+        ----------
+        banner : typing.Optional[builtins.str]
+            The package to find a `banner.txt` in.
+        allow_color : builtins.bool
+            A flag that allows advising whether to allow color if supported or
+            not. Can be overridden by setting a `"CLICOLOR"` environment
+            variable to a non-`"0"` string.
+        force_color : builtins.bool
+            A flag that allows forcing color to always be output, even if the
+            terminal device may not support it. Setting the `"CLICOLOR_FORCE"`
+            environment variable to a non-`"0"` string will override this.
+
+        !!! note
+            `force_color` will always take precedence over `allow_color`.
+        """
+        ux.print_banner(banner, allow_color, force_color)
+
+    async def close(self) -> None:
+        # self._server.start should raise errors.ComponentStateConflictError if it's not running
+        await self._server.close()
+        await self._rest.close()
+
+    async def join(self) -> None:
+        await self._server.join()
+
+    async def on_interaction(self, body: bytes, signature: bytes, timestamp: bytes) -> interaction_server_.Response:
+        return await self._server.on_interaction(body, signature, timestamp)
+
+    def run(
+        self,
+        asyncio_debug: typing.Optional[bool] = None,
```

Comment:
```suggestion
        asyncio_debug: bool = False,
```

Reply:
this is copy pasted from the gateway bot impl

Reply:
Should be changed on both then. Makes no sense to allow for `None`. I can open another pr if you want